fluid/PaddleCV/image_classification/dist_train/dist_train.py

```diff
@@ -314,73 +230,65 @@ def train_parallel(train_args, test_args, args, train_prog, test_prog,
         build_strategy.reduce_strategy = fluid.BuildStrategy(
         ).ReduceStrategy.AllReduce
 
-    avg_loss = train_args[0]
-
-    if args.update_method == "pserver":
+    if args.update_method == "pserver" or args.update_method == "local":
         # parameter server mode distributed training, merge
         # gradients on local server, do not initialize
         # ParallelExecutor with multi server all-reduce mode.
         num_trainers = 1
         trainer_id = 0
+    else:
+        num_trainers = args.dist_env["num_trainers"]
+        trainer_id = args.dist_env["trainer_id"]
 
     exe = fluid.ParallelExecutor(
         True,
-        avg_loss.name,
+        train_cost.name,
         main_program=train_prog,
         exec_strategy=strategy,
         build_strategy=build_strategy,
         num_trainers=num_trainers,
         trainer_id=trainer_id)
 
-    pyreader = train_args[4]
-    for pass_id in range(args.pass_num):
+    over_all_start = time.time()
+    fetch_list = [train_cost.name, train_acc1.name, train_acc5.name]
+    for pass_id in range(args.num_epochs):
         num_samples = 0
         start_time = time.time()
-        batch_id = 0
-        pyreader.start()
+        batch_id = 1
+        train_pyreader.start()
         while True:
```

Comment:
`while True => for step_id in range(total_steps)` to avoid nccl hung at the last batch of the last epoch.